### PR TITLE
Fix return code from write_pcre

### PR DESCRIPTION
**Pull request by kamahen**

### CODE REVIEW

File: pcre4pl.c [hunk 1]
```diff
@@ -247,11 +247,12 @@ static int
...
 { (void)flags; /* unused arg */
   const re_data *re = PL_blob_data(symbol, NULL, NULL);
+  int rc;
   /* For blob details: re_portray_() - re_portray/2 */
   PL_STRINGS_MARK();
-  SfprintfX(s, "<regex>(%p, /%Ws/)", re, PL_atom_wchars(re->pattern, NULL));
+  rc = SfprintfX(s, "<regex>(%p, /%Ws/)", re, PL_atom_wchars(re->pattern, NULL));
   PL_STRINGS_RELEASE();
-  return TRUE;
+  return rc > 0;
```
Comment (JanWielemaker) [inline pcre4pl.c:255]: There is no need for this.     If something goes wrong the stream is in an error state and that will be picked up.

Comment (kamahen) [reply pcre4pl.c:255]: It is not documented that whatever calls the write() callback checks the streams' error state. (Or is this done by subsequent I/O operations on the stream that check the error state and (eventually) report the error?)
Also, it's good style to check the return codes of I/O functions.

Comment (kamahen) [reply pcre4pl.c:255]: If I had two Sfprintf()s, the first should have a check for non-negative return code, I think. So, it's only in the case of a write callback containing a single Sfprintf() that no rc-checking is needed. This seems error-prone ...

[Also, I suppose I should add documentation that the write callback is in the context of a PL_release_stream(), which does a check; and also add that to PL_release_stream()'s documentation?]

Perhaps I should add Sfprintf() et al wrappers to SWI-cpp-plx.h, the way I've done for the PL_*() functions? This won't help package/pcre (it's still in C, not C++), but would help with packs written in C++.

Comment (kamahen) [reply pcre4pl.c:255]: OK, I'll update the documentation accordingly and remove this PR when I'm done.

Comment (JanWielemaker) [reply pcre4pl.c:255]: Thanks.   Pushed SWI-Prolog/swipl-devel.git@d49b1d79a0a5a83a3c7f1f53a9ed5a4e257ad367 to make the failure immediate and explicit.

**Comment by kamahen** [comment]
I've added a bit of documentation in https://github.com/SWI-Prolog/swipl-devel/pull/1191

**Pull request closed without merging**